engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/function/ApplicationOperator.java

```diff
@@ -26,8 +26,8 @@ public class ApplicationOperator extends Node {
     invokeCallableNode.setTailStatus(BaseNode.TailStatus.TAIL_DIRECT);
   }
 
-  Stateful execute(
-      VirtualFrame frame, @MonadicState Object state, Object self, @Suspend Object argument) {
+  Object execute(
+      VirtualFrame frame, @MonadicState State state, Object self, @Suspend Object argument) {
```

Comment:
Do we need the `@MonadicState` annotation? Isn't it enough to check for `State` and treat each `State` as a carrier for the `@MonadicState`?